### PR TITLE
NRPT-711: Update redacted record subset on mongoose save and update

### PR DESCRIPTION
**Pull request by MaxWardle**
[NRPT-711](https://bcmines.atlassian.net/browse/NRPT-711)

### CODE REVIEW

File: api/materialized_views/search/redactedRecordSubset.js [hunk 1]
```diff
@@ -75,3 +75,26 @@ async function updateOneRecord(record) {
...
+ *
+ * @param {*} record
+ */
+async function deleteOneRecord(record) {
+  try {
+    defaultLog.info('Updating redacted_record_subset');
+
+    const db = mongodb.connection.db(process.env.MONGODB_DATABASE || 'nrpti-dev');
+    const redactedCollection = db.collection('redacted_record_subset');
+    redactedCollection.deleteOne({ _id: record._id });
```
Comment (marklise) [inline api/materialized_views/search/redactedRecordSubset.js:92]: `deleteOne` returns a promise

File: api/materialized_views/search/redactedRecordSubset.js [hunk 2]
```diff
@@ -1,244 +1,100 @@
...
+
+  const redactedRecord = redactRecord(record);
 
   try {
+    defaultLog.info('Updating redacted_record_subset');
+
     const db = mongodb.connection.db(process.env.MONGODB_DATABASE || 'nrpti-dev');
-    const mainCollection = db.collection('nrpti');
+    const redactedCollection = db.collection('redacted_record_subset');
+    redactedCollection.save(redactedRecord);
```
Comment (marklise) [inline api/materialized_views/search/redactedRecordSubset.js:43]: `await` here

File: api/materialized_views/search/redactedRecordSubset.js [hunk 3]
```diff
@@ -1,244 +1,100 @@
...
-        }
-      }
-    });
-
-    aggregate.push({ $out: 'redacted_record_subset' });
-
-    await mainCollection.aggregate(aggregate).next();
+    const db = mongodb.connection.db(process.env.MONGODB_DATABASE || 'nrpti-dev');
+    const redactedCollection = db.collection('redacted_record_subset');
+    redactedCollection.findOneAndUpdate({ _id: redactedRecord._id }, redactedRecord);
```
Comment (marklise) [inline api/materialized_views/search/redactedRecordSubset.js:69]: `await`

**Review by mtCarto — APPROVED** [review]
Looks good, confirmed that the redacted subset is properly updated on record changes 👍

**Review by BcGovNeal — APPROVED** [review]

**Review by marklise — APPROVED** [review]

**Pull request merged**